src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -2974,6 +2982,18 @@ static PaError GetClosestFormat(IAudioClient *client, double sampleRate, const P
 
     if (hr == S_OK)
     {
+        // Workaround for the Realtek driver bug (see https://github.com/PortAudio/portaudio/issues/875):
```

Comment:
It would be nice to gate this change based on whether the driver is Realtek and on the version.
That would make it less likely to cause a regression on systems that do not have the Realtek bug.

Reply:
Yes, it can be gated by device name as all Realtek devices have "Realtek" string inside the name, so it can be:

```
if (strstr(device_name, "Realtek") != NULL)
{
    // workaround action goes here
}
```

Reply:
@philburk I implemented your proposal, would you please check.

Reply:
Thanks.